### PR TITLE
Adds Morph - non-ling shapeshifter

### PR DESCRIPTION
**Pull request by AnturK**
This was intended as pricey changeling ability but i felt it would need to be more integrated and that's too much work. And mini-antag mobs are all the rage now. Still unfinished.
### How does it work:

Examine anything to assume it's shape. 5 second cooldown between uses.
While shapeshifted you're fast but weak, strong but slow in base form.
Examine can reveal the trick if examiner is close enough.
Can ventcrawl and shapeshift in the pipes.
### Todo
- [x] Add /screen object checks and other stuff you shouldn't impersonate (singulo mimic etc)
- [x] Find proper sprite
- [x] Add player help / spawn buttons
- [x] Remove from xenobio lists / make it /hostile

### CODE REVIEW

**Comment by RemieRichards** [comment]
morph

Morph.

**Comment by Iamgoofball** [comment]
what is his goal

what does he do

how can he harm the station and it's crew

**Comment by Dorsisdwarf** [comment]
The morph's objectives should be something disruptive but not too bad. Like, it would be a case of "Wait wh- oh, a fucking morph did it?" /"But I was in medbay officer!"/ ";beep boop jack steiner toolboxed an officer setting to arrest" / "Where the tits did my donuts go?"/

**Comment by AnturK** [comment]
A) Nothing for now. Will add some specific goals like maybe eating Ian or stealing food.
B) Turn into Beepsky and eat people.
C) By eating people who do not suspect it.

Seriously though, i'll be probably adding some simple ai for npc version so it's more mimic-y than our current ones. We lack true shapeshifter enemies and wanted to fill that niche.

**Comment by Dorsisdwarf** [comment]
Wait, when you say "examine *anything* "...

**Comment by RemieRichards** [comment]
@Dorsisdwarf yes, it can copy anything.

**Comment by AnturK** [comment]
Anything but turfs. So no floor mimics. And will be adding some blacklist soon.

File: code/modules/mob/living/simple_animal/morph/morph.dm [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	if(morph_time <= world.time)
+		if(A == src)
+			restore()
+			return
+		if(istype(A,/atom/movable/))
+			assume(A)
+	else
+		..()
+
+/mob/living/simple_animal/morph/proc/assume(var/atom/target)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/morph/morph.dm]: change this to ```var/atom/movable/target```
then remove the check in ```ShiftClickOn()```
then add an ```if(!istype(target))``` to ```assume()```

File: code/modules/mob/living/simple_animal/morph/morph.dm [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	maxHealth = 150
+	health = 150
+	environment_smash = 1
+	melee_damage_lower = 30
+	melee_damage_upper = 30
+	see_in_dark = 8
+	see_invisible = SEE_INVISIBLE_MINIMUM
+
+	var/morphed = 0
+	var/atom/form = null
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/morph/morph.dm]: ```var/atom/movable/form```

File: code/modules/mob/living/simple_animal/morph/morph.dm (hunk 1 above)
Comment (AnturK) [inline code/modules/mob/living/simple_animal/morph/morph.dm]: Oh right. Forgot to swap it back when i made it more specific.

**Comment by WJohn** [comment]
This sounds exceedingly difficult to play without being a cig butt every time. Perhaps have your health and damage scale with the size of the object you've assumed?

**Comment by Xhuis** [comment]
ditto

**Comment by Iamgoofball** [comment]
Prop Hunt: The Antag

**Comment by PKPenguin321** [comment]
\>can't disguise as the singulo

why bother playing as it then???

kidding, it sounds cool :+1:

**Comment by Jalleo** [comment]
Toy singulo it is

**Comment by TheVekter** [comment]
"Why does that burger have teeth?"
mimics fuck year

**Comment by SconesC** [comment]
Oh, so it's just mimics? Awesome.

**Comment by AnturK** [comment]
Merge ready. I left the objectives alone for now since it's admin only anyway.

**Comment by Man-Shroom** [comment]
I like this. This can embody the more stealthy-esque aspects of ling, and my (eventual) rewrite of changeling can handle the 'rest'. 

(It's totally going to arrive this summer you guys I swear.)

**Comment by Miauw62** [comment]
its not, because its shit

**Comment by Iamgoofball** [comment]
why is it shit

**Comment by AnturK** [comment]
So, is this bad. Say so i can close this. 2 weeks since it's done already with no response.

**Comment by TheVekter** [comment]
Don't. I think it's cool and @Miauw62 is just PMSing, unless he wants to expand on why he doesn't like it.

**Comment by Miauw62** [comment]
I just arrived at a bed and breakfast in France, didn't really have the
time to merge it. I'll do so now.
Op 13-jul.-2015 11:38 schreef "TheVekter" <notifications@github.com>:

**Comment by Iamgoofball** [comment]
>france

**Pull request merged**